### PR TITLE
chore: fixed failing builds, updated docker login

### PR DESCRIPTION
**Pull request by 34fathombelow**
Signed-off-by: Justin Marquis <34fathombelow@protonmail.com>

Fixed the image build failing for push events. Added `GOLANG_VERSION` as an environment variable.  In PR#499 changes were made in the Makefile, they are still needed for `github/workflows/ci-build.yaml` left them as is.

### CODE REVIEW

**Comment by 34fathombelow** [comment]
Also fixes issue #515

**Review by crenshaw-dev — COMMENTED** [review]
Overall, LGTM! Added one question for Alex.

File: .github/workflows/image.yaml
Comment (34fathombelow) [reply .github/workflows/image.yaml:44]: Please hold off on this PR.  I believe I have a better solution.  Should I submit a new PR?

**Comment by 34fathombelow** [comment]
Please review the last commit, I believe that this is a better solution than my first commit.  I reverted a change in Makefile from PR #499 , to provide backwards compatibility for single architecture builds.  While adding `multi-images` to the Makefile for multi-architecture builds.

Set a default variable `DOCKER_PUSH=false` setting `DOCKER_PUSH=true` will cause a build to push to a registry.   We can now use `make muti-images` for future releases.

File: .github/workflows/image.yaml
Comment (crenshaw-dev) [inline .github/workflows/image.yaml:50]: One option would be to hard-code the buildx command, mimicking what's done in Argo CD. https://github.com/argoproj/argo-cd/blob/7cf6a5b1907b8b16b57e419a52b4c34ed8c74a55/.github/workflows/image.yaml#L49

But I'm alright with the new Make target too.

**Review by crenshaw-dev — APPROVED** [review]
LGTM!

File: .github/workflows/image.yaml
Comment (34fathombelow) [reply .github/workflows/image.yaml:50]: I'm fine with the hardcoded build command,  but was not sure if the phony target was needed.  Let me know what path you would like to take.   Whatever makes the devs life easier.

Comment (34fathombelow) [reply .github/workflows/image.yaml:50]: Sounds great,  thanks for all your help.

**Comment by oleg-glushak** [comment]
Would be great to have this merged.

**Comment by crenshaw-dev** [comment]
@oleg-glushak there's a batch of AppSet PRs that are getting a little old. I'm going to ping some folks for reviews on them today. Thanks for your patience!

**Review by mikevader — APPROVED** [review]

**Comment by mikevader** [comment]
What is holding back this PR? Is some help required?

If I see it correctly, this is the only thing preventing applicationset to support ARM64, and as ArgoCD is support it officially since 2.3 it would be really sweet to finally archive our custom build infra.

**Comment by crenshaw-dev** [comment]
@mikevader we're a bit short on ApplicationSet approvers at the moment. The situation should improve as we integrate AppSet more closely with Argo CD. 

For the moment, I've requested approval permissions on this repo since I already have approval permissions on Argo CD.

**Comment by rishabh625** [comment]
@crenshaw-dev : I guess this is not required as with [merge](https://github.com/argoproj/argo-cd/pull/8864) we have support

**Comment by crenshaw-dev** [comment]
Yep, thanks again for this work @34fathombelow!

**Pull request closed without merging**
